functions/createReview/astParsing/index.js

```diff
@@ -0,0 +1,101 @@
+import { Parser } from 'acorn'
+import tsPlugin from 'acorn-typescript'
+import path from 'node:path'
+import { LooseParser } from 'acorn-loose'
+import jsxPlugin from 'acorn-jsx'
+
+import validators from './rules/index.js'
+import { parseAndAggregate } from './parser.js'
+import { mapLineToDiff } from 'map-line-to-diff'
+
+function getParser(fileName) {
+  const fileExtension = path.extname(fileName)
+  let parser
+  if (fileExtension === '.ts' || fileExtension === '.tsx') {
+    parser = Parser.extend(tsPlugin())
+  } else if (fileExtension === '.jsx') {
+    parser = Parser.extend(jsxPlugin())
+  } else if (fileExtension === '.js') {
+    parser = Parser
+  }
+  return parser
+}
+
+export function isESMFile(fileContent) {
+  return (
+    /^\s*import\s+/gm.test(fileContent) || /^\s*export\s+/gm.test(fileContent)
```

Comment:
wondering if there's a better way to test this. not so much because it's not nice, but because it feels it's quite slow

Reply:
I can think of two alternatives that don't strike me as faster:
1. Use the `acorn-loose` AST parser to parse the code and then reparse the file with the correct sourceType.
2. Look for the closest package.json file if esm is being used. The additional http call to github might be faster than text parsing all the files.